metricbeat/module/vsphere/datastorecluster/datastorecluster.go

```diff
@@ -84,18 +94,45 @@ func (m *DatastoreClusterMetricSet) Fetch(ctx context.Context, reporter mb.Repor
 	}()
 
 	var datastoreCluster []mo.StoragePod
-	err = v.Retrieve(ctx, []string{"StoragePod"}, nil, &datastoreCluster)
+	err = v.Retrieve(ctx, []string{"StoragePod"}, []string{"name", "summary", "childEntity"}, &datastoreCluster)
 	if err != nil {
 		return fmt.Errorf("error in retrieve from vsphere: %w", err)
 	}
 
+	pc := property.DefaultCollector(c)
 	for i := range datastoreCluster {
 		if ctx.Err() != nil {
 			return ctx.Err()
 		}
 
-		reporter.Event(mb.Event{MetricSetFields: m.mapEvent(datastoreCluster[i])})
+		assetNames, err := getAssetNames(ctx, pc, &datastoreCluster[i])
+		if err != nil {
+			m.Logger().Errorf("Failed to retrieve object from host %s: %w", datastoreCluster[i].Name, err)
```

Comment:
In case of an error, do we still report an event on L#113?

Reply:
Yes, we will still push events with remaining metrics.
These assets are extra information that we attach manually for user reference.